profiles/common/modules/custom/ecas/includes/ecas.inc

```diff
@@ -441,133 +506,170 @@ function ecas_login_check() {
 
-    // Also set an extra parameter for ecas_sync_user_info() -- this allows
-    // other modules to detect a user creation.
-    $ecas_sync_user_info_args['ecas_user_creation'] = TRUE;
+  // Also set an extra parameter for ecas_sync_user_info() -- this allows
+  // other modules to detect a user creation.
+  $ecas_sync_user_info_args['ecas_user_creation'] = $_SESSION['ecas_first_login'];
+
+  // Synchronize data between Drupal user object and Ecas ones according to the
+  // site specificity.
+  // There is no save inside, that will be do after further validations.
+  $account_edit = ecas_sync_drupal_user_with_ecas_info($account, $ecas_attributes, $ecas_sync_user_info_args);
+
+  // Before saving and as we gave a last chance to other modules to change the
+  // user data, check the validity of the user that must be saved.
+  // The ecas user must have a defined e-mail; otherwise the drupal
+  // authentication is denied.
+  if (empty($account_edit['mail'])) {
+    _ecas_warning_goto(
+      ECAS_WARNING_REASON_NO_EMAIL,
+      $ecas_user_language,
+      'Login denied for %ecas_name because it has no e-mail defined.',
+      array(
+        '%ecas_name' => $ecas_name,
+      )
+    );
   }
 
-  // Final check to make sure we have a good user.
-  if ($account && $account->uid > 0) {
-    // Retrieve information about the user from ECAS attributes.
-    $user_info = phpCAS::getAttributes();
+  // Check that the email is not used for another account; otherwise the drupal
+  // authentication is denied.
+  if ($ecas_social_login == 'v' && ecas_is_email_already_used($account_edit['mail'], $account)) {
+    _ecas_warning_goto(
+      ECAS_WARNING_REASON_EXISTING_EMAIL,
+      $ecas_user_language,
+      'Login denied for %ecas_name because the e-mail is already used by another account.',
+      array(
+        '%ecas_name' => $ecas_name,
+      )
+    );
+  }
 
-    // Update her information.
-    if (!empty($user_info)) {
-      ecas_sync_user_info($account, $user_info, $ecas_sync_user_info_args);
-    }
-    else {
-      $args = array(':username' => $account->name);
-      drupal_set_message(t('Warning: no information could be found for username :username', $args));
-    }
+  // Check that social login user has authenticated role only.
+  $authenticated_role = user_role_load_by_name('authenticated user');
 
-    // Case when account is blocked.
-    if ($account->status == 0) {
-      _ecas_session_end();
-      drupal_goto('ecas_warning_page');
-    }
-    // When returning with same ECAS login as before closing Drupal session.
-    else {
-      if (
-        isset($account->data['ecas_login_date']) &&
-        $account->data['ecas_login_date'] == $user_info['cas:loginDate']
-      ) {
-
-        $msg = 'Rejected ECAS reused login: %n (ECAS)';
-        watchdog('user', $msg, array('%n' => $account->name), WATCHDOG_NOTICE);
+  if ($ecas_social_login == 'v' && (!user_has_role($authenticated_role->rid, $account) || count($account->roles) > 1)) {
```

Comment:
This condition will prevent users from signing up with a Social Media account. As the account has not yet been created, it will not have the Authenticated User role and will get the error message.